Sources/Navidux/Core/Navigation/NavigationAction/Actions/PushNavigationAction.swift

```diff
@@ -29,13 +37,34 @@ public struct PushNavigationAction: NavigationAction {
         case .fullScreen:
             coordinator.navigationController.pushViewController(screen, animated: self.animated)
         case .modal(let completion):
+            coordinator.store.hasOverlay = true
+            coordinator.navigationController.present(screen, animated: self.animated, completion: completion)
+        case .bottomSheet(let size, let completion):
+            switch size {
+            case .auto:
+                coordinator.bottomSheetTransitioningDelegate.sheetSize = .auto
+                screen.transitioningDelegate = coordinator.bottomSheetTransitioningDelegate
+                screen.modalPresentationStyle = .custom
+            case .fixed(let height):
+                coordinator.bottomSheetTransitioningDelegate.sheetSize = .fixed(height)
+                screen.transitioningDelegate = coordinator.bottomSheetTransitioningDelegate
+                screen.modalPresentationStyle = .custom
+            case .halfScreen:
+                coordinator.bottomSheetTransitioningDelegate.sheetSize = .halfScreen
+                screen.transitioningDelegate = coordinator.bottomSheetTransitioningDelegate
+                screen.modalPresentationStyle = .custom
+            case .fullScreen:
+                screen.modalPresentationStyle = .formSheet
+            }
+            
+            coordinator.store.hasOverlay = true
             coordinator.navigationController.present(screen, animated: self.animated, completion: completion)
```

Comment:
It would be better to have this logic incapsulated in some entity. Maybe we should introduce `BSBuilder` and feed all the values there

Reply:
Done